IPython/lib/display.py

```diff
@@ -383,36 +383,41 @@ def __init__(self,
         result_html_suffix : str
             text to append at the end of link [default: '<br>']
         """
-        if isdir(path):
+        if Path(path).is_dir():
             raise ValueError("Cannot display a directory using FileLink. "
               "Use FileLinks to display '%s'." % path)
-        self.path = fsdecode(path)
+        self.path = Path(fsdecode(path))
```

Comment:
This is where the problem starts. The path which is sent is in the form `/var/T/.........` whereas `self.path` gets or changes to the form `/private/var/T/......` in the build.

Reply:
That's my best guess too. Could you try reverting to the old version so we can confirm that?

Reply:
I am bringing back the `abspath` method for now. Will try to remove it later.

Reply:
The [MacOS build](https://travis-ci.com/github/ipython/ipython/jobs/415283806) actually didn't fail after reverting to the old version.

Reply:
`Path.absolute` works great in place of `abspath`.

Reply:
So i looked up a little and found out that `resolve()` not only normalizes the path but also resolves the symlinks, if any(hence the name) which causes the problem. So should i change the test, which fails because of the "resolve symlinks" part, to
```
expected = str(pathlib.Path(tf.name).resolve())
```
?

Reply:
Great detective work!
> But one problem of absolute() is that it doesn't normalize the path. So should i change the test, which fails because of the "resolve symlinks" part
I will leave that decision to the maintainers

Reply:
> So i looked up a little and found out that `resolve()` not only normalizes the path but also resolves the symlinks, if any(hence the name) which causes the problem. But one problem of `absolute()` is that it doesn't normalize the path. So should i change the test, which fails because of the "resolve symlinks" part, to
> ```
> expected = str(pathlib.Path(tf.name).resolve())
> ```
> ?

@Carreau should I do this?